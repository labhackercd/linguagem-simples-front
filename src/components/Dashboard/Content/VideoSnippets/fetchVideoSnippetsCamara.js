import axiosInstance from '../../../../auth/axiosApi'
import {API_VIDEOS_SNIPPETS} from '../../../../api_urls'


export default async function fetchSessionVideos(sessionIdDadosAbertos){
    const response = await axiosInstance.get((API_VIDEOS_SNIPPETS+sessionIdDadosAbertos), {});

    if(response.status===200){
        return response.data;
    }else{
        return null;
    }
}
