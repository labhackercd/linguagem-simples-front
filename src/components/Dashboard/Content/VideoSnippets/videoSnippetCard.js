import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import ButtonBase from '@material-ui/core/ButtonBase'
import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'

export default function SnippetCard(props){
  const deputado = props.data.author;
  const siglaPartidaria = props.data.legend;

  const [open, setOpen] = React.useState(false);


  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = (value) => {
    setOpen(false);
  };

  return (
    <Box width="100%" height="100%" marginX={0.3} marginTop={0.5}>
      <ButtonBase onClick={handleClickOpen} id={"openModal"}>
        <div>
            <img style={{width:"100%", height:"6vh"}} id="image" src={props.data.thumbnail} alt="Thumbanail trecho"></img>
            <Box fontSize={10}>
            <Typography>{deputado}</Typography>
            </Box>
            <Box fontSize={8}>
            <Typography variant="body2">{siglaPartidaria}</Typography>
            </Box>
        </div>
        </ButtonBase>
      <VideoSnippetModal open={open} onClose={handleClose} data={props.data} />
    </Box>
  );
}