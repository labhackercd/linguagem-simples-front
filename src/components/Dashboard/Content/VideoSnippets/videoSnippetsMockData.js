 export const videoSnippetsMockData = [
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234414000",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h53'34\"\nduração - 00:00:58",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234452950",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h54'12\"\nduração - 00:00:59",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234492497",
    "description": "Paulão\n                                                                        \nDeputado PT-AL\n\nhorário - 15h54'52\"\nduração - 00:02:33",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/171617.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234625877",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h57'05\"\nduração - 00:00:34",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
    "description": "Arthur Oliveira Maia\n                                                                        \nDeputado DEM-BA\n\nhorário - 15h57'20\"\nduração - 00:02:04",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234744800",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h59'04\"\nduração - 00:01:56",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234841110",
    "description": "Alessandro Molon\n                                                                        \nDeputado PSB-RJ\n\nhorário - 16h00'41\"\nduração - 00:00:52",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160511.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234874013",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h01'14\"\nduração - 00:01:14",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234928823",
    "description": "Paulão\n                                                                        \nDeputado PT-AL\n\nhorário - 16h02'08\"\nduração - 00:01:02",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/171617.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234971410",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h02'51\"\nduração - 00:00:34",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234986130",
    "description": "Fernanda Melchionna\n                                                                        \nDeputado PSOL-RS\n\nhorário - 16h03'06\"\nduração - 00:01:08",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/204407.jpg"
  },
  {
    "id": 12,
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594235034203",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h03'54\"\nduração - 00:01:21",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "id": 13,
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594235095990",
    "description": "Arnaldo Jardim\n                                                                        \nDeputado CIDADANIA-SP\n\nhorário - 16h04'55\"\nduração - 00:00:22",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/141391.jpg"
  }
]