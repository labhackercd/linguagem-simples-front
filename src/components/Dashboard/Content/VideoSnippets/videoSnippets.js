import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import CircularProgress from '@material-ui/core/CircularProgress';
import {videoSnippetsMockData} from './videoSnippetsMockData'
import fetchSessionVideos from './fetchVideoSnippetsCamara'
import { FixedSizeList } from 'react-window';
import ListItem from '@material-ui/core/ListItem';
import InputAdornment from '@material-ui/core/InputAdornment';
import SearchIcon from '@material-ui/icons/Search';
import TextField from '@material-ui/core/TextField';
import ButtonBase from '@material-ui/core/ButtonBase'

import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'
import SnippetCard from './videoSnippetCard'



export default class VideoSnippets extends React.Component {
    
  constructor(props){
    super(props);
    this.state = { 
        snippets: '',
        filteredSnippets: '',
        sessionHasStarted: false,
        dataLoaded: false,
        searchField: '',
        error:{
          happened:false,
          message: "Erro",
          status:""
        }
    };
  }

  componentDidMount(){
      this._isMounted = true;
      //console.log(this.props)
      if(this._isMounted){

          if(this.props.sessionInfo.situation_session === "pre_session"){
            this.setState({sessionHasStarted:false});
            //this.setState({snippets:true});
          }else{
            this.setState({sessionHasStarted:true})
            this.fetchSessionsList();
          }
          
      }
  }

  fetchSessionsList = async term => {
    try {
      //console.log(this.props)
      const data = await fetchSessionVideos(this.props.sessionInfo.id_session_dados_abertos);
      this.setState({snippets:data})
      this.setState({filteredSnippets:data})
      this.setState({dataLoaded:true});
    } catch (error) {
      //console.log(error)
      this.setState({error:{happened:true,status:error.response.status} })
    }
  };

  renderSearchBarFunction(){
    return(
      <React.Fragment>
        <Grid container>
          <Grid item xs={10}>
            <Box marginLeft={2} marginTop={1}>
              <Typography variant="h5" style={{ color: "#3E3E3E" }}>Trechos</Typography>
            </Box>
          </Grid>
          <Grid item xs={2}>
            <Box marginRight={1}> 
              <TextField
                id="input-search-snippet"
                size="small"
                onChange={this.snippetsFilterOnChange}
                InputProps={{
                  endAdornment: (<InputAdornment position="start"><SearchIcon /></InputAdornment>),
                }}/>
            </Box>
          </Grid>
        </Grid>
      </React.Fragment>
    )
  }

  renderListItem = ({index, style}) => {
    return(
      <ListItem disableGutters={true} style={style} >
          <SnippetCard data = {this.state.filteredSnippets[index]}></SnippetCard>        
      </ListItem>
    )
  }

  snippetsFilterOnChange = (event) => {
    //console.log(event.target.value)
    this.setState({
      searchField: event.target.value
    })
    
    this.setState({
      filteredSnippets: this.state.snippets.filter(term => term.author.toLowerCase().includes(this.state.searchField.toLowerCase()))
    })
    
  }

  render(){
    const widthSnippetsBox = parseInt(((window.innerWidth)*0.54));
    const widthSnippetsItem = (((window.innerWidth)*0.5)*0.12);
    //console.log(widthSnippetsBox)
   
    if(this.state.error.happened){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>Erro</Typography>
              </Box>)
    }

    if(!this.state.sessionHasStarted){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>A transmissão ainda não foi iniciada.</Typography>
              </Box>)
    }

    if(!this.state.dataLoaded){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}><CircularProgress></CircularProgress></Box>)
    }

    return (
      <div>
        <Grid container >
        {this.renderSearchBarFunction()}
          <Grid item xs={12}>
            <Box marginLeft={1}>
              <FixedSizeList width={widthSnippetsBox} height={"12.5vh"} itemSize={widthSnippetsItem} layout="horizontal" itemCount={this.state.filteredSnippets.length}>        
                {this.renderListItem}
              </FixedSizeList>
            </Box>
          </Grid>
        </Grid>
      </div>
    )
  }
}