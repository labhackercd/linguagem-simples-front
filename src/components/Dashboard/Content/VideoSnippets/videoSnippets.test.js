import React from 'react';
import { render, screen } from '@testing-library/react';
import {shallow} from "enzyme/build";
import VideoSnippets from './videoSnippets'

it("should render the VideoSnippets section", () => {
    const component = shallow(<VideoSnippets/>);

    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});
