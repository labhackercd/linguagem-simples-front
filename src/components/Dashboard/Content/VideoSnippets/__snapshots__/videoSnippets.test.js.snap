// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the VideoSnippets section 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={10}
      >
        <Styled(MuiBox)
          marginLeft={2}
          marginTop={1}
        >
          <WithStyles(ForwardRef(Typography))
            style={
              Object {
                "color": "#3E3E3E",
              }
            }
            variant="h5"
          >
            Trechos
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <Styled(MuiBox)
          marginRight={1}
        >
          <WithStyles(ForwardRef(TextField))
            InputProps={
              Object {
                "endAdornment": <ForwardRef(WithStyles)
                  position="start"
                >
                  <UNDEFINED />
                </ForwardRef(WithStyles)>,
              }
            }
            id="input-search-snippet"
            onChange={[Function]}
            size="small"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Styled(MuiBox)
        marginLeft={1}
      >
        <List
          direction="ltr"
          height="12vh"
          itemCount={13}
          itemSize={61.44}
          layout="horizontal"
          overscanCount={2}
          useIsScrolling={false}
          width={552.96}
        >
          <Component />
        </List>
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
